agate/data_types/number.py

```diff
@@ -94,7 +94,7 @@ def cast(self, d):
 
         try:
             return Decimal(d) * sign
-        except InvalidOperation:
+        except (InvalidOperation, ValueError):
```

Comment:
Can you explain why `ValueError` is needed here?

Reply:
Because PyPy3 errors like this when `d` is some random Unicode char:
https://travis-ci.org/lcorbasson/agate/jobs/645585144

```
======================================================================
ERROR: test_test (tests.test_data_types.TestNumber)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/python/pypy3.6-7.1.1/lib_pypy/_decimal.py", line 153, in _handle_decimaldigits
    res += str(_unicodedata.digit(c))
ValueError: not a digit
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/home/travis/build/lcorbasson/agate/tests/test_data_types.py", line 142, in test_test
    self.assertEqual(self.type.test(u'⇔'), False)
  File "/home/travis/build/lcorbasson/agate/agate/data_types/base.py", line 29, in test
    self.cast(d)
  File "/home/travis/build/lcorbasson/agate/agate/data_types/number.py", line 96, in cast
    return Decimal(d) * sign
  File "/opt/python/pypy3.6-7.1.1/lib_pypy/_decimal.py", line 175, in __new__
    return cls._from_object(value, context, exact=True)
  File "/opt/python/pypy3.6-7.1.1/lib_pypy/_decimal.py", line 198, in _from_object
    return cls._from_str(value, context, exact=exact, strip=exact)
  File "/opt/python/pypy3.6-7.1.1/lib_pypy/_decimal.py", line 238, in _from_str
    s = str.encode(value, 'ascii', '_decimal_encode')
  File "/opt/python/pypy3.6-7.1.1/lib_pypy/_decimal.py", line 153, in _handle_decimaldigits
    res += str(_unicodedata.digit(c))
ValueError: encoding with 'ascii' codec failed (ValueError: not a digit)
----------------------------------------------------------------------
```

Reply:
I added a comment to explain this